### PR TITLE
Setup with nx after better understanding its framwork

### PR DESCRIPTION
**Pull request by unicop**
This PR include:
- Use nx as an executor when possible rather than package.json
- rename migrate-to-typescript plugin into a shorter name for convinient while developing (the npm package will still call the same name)
- Setup docusaurus through nx with @nx-plus/docusaruus executor (Added the left commands to the package.json for now)

### CODE REVIEW

File: apps/website/src/components/HomepageFeatures/index.tsx [hunk 1]
```diff
@@ -1,61 +1,67 @@
...
   {
-    title: 'Powered by Community',
-    Svg: require('@site/static/img/undraw_docusaurus_react.svg').default,
+    title: "Powered by Community",
+    Svg: require("@site/static/img/undraw_docusaurus_react.svg").default,
     description: (
       <>
-        ESLint is consistently getting <a href='https://www.npmtrends.com/eslint-vs-prettier-vs-tslint'>popularity</a>, We have researched the whole eco-system to find the best 
-        recipe we can to help spreading standards that can be easily understood
+        ESLint is consistently getting{" "}
```
Comment (neriyarden) [inline apps/website/src/components/HomepageFeatures/index.tsx:41]: i think "*gaining* popularity" or "getting *popular*" is more accurate

File: tools/scripts/publish.mjs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const validVersion = /^\d+\.\d+\.\d+(-\w+\.\d+)?/;
+invariant(
+  version && validVersion.test(version),
+  `No version provided or version did not match Semantic Versioning, expected: #.#.#-tag.# or #.#.#, got ${version}.`
+);
+
+const graph = readCachedProjectGraph();
+const project = graph.nodes[name];
+
+invariant(
```
Comment (neriyarden) [inline tools/scripts/publish.mjs:36]: what does "invariant" mean? is it like "invalid"?

Comment (neriyarden) [reply tools/scripts/publish.mjs:36]: anyway i like this function :)

**Review by neriyarden — APPROVED** [review]

File: apps/website/docusaurus.config.js [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
+    "An opinionated set of ESLint extensions that will help you spread high-standards",
+  url: "https://uniqorg.github.io",
+  baseUrl: "/eslint-extensions/",
+  onBrokenLinks: "throw",
+  onBrokenMarkdownLinks: "warn",
+  favicon: "img/eslint-favicon.512x512.png",
 
   // GitHub pages deployment config.
   // If you aren't using GitHub pages, you don't need these.
-  organizationName: 'uniqorg', // Usually your GitHub org/user name.
```
[the remark this answers is not in the document]
Comment (unicop) [reply apps/website/docusaurus.config.js:18]: for now, we do not have a lint configuration for this project, we need to define one and then fix it, so I'm not sure why it did it, but it is not critical for now

File: tools/scripts/publish.mjs (hunk 2 above)
Comment (unicop) [reply tools/scripts/publish.mjs:36]: this script/function was generated by nx to publish packages, so I'm not sure, although I did a quick brief on it, and understood it approximately

**Pull request merged**